rest/blip_client_test.go

```diff
@@ -1135,3 +1143,60 @@ func (btc *BlipTesterCollectionClient) sendPushMsg(msg *blip.Message) error {
 	btc.addCollectionProperty(msg)
 	return btc.parent.pushReplication.sendMsg(msg)
 }
+
+func (btc *BlipTesterClient) BlipClientCollectionSetup(collection *db.DatabaseCollection) (*BlipTesterCollectionClient, error) {
+	var btcCollection *BlipTesterCollectionClient
+	scopeAndCollectionKey := strings.Join([]string{collection.ScopeName(), collection.Name()}, base.ScopeCollectionSeparator)
+	if !base.IsDefaultCollection(collection.ScopeName(), collection.Name()) {
+		checkpointID1 := "checkpoint1"
+		getCollectionsRequest, err := db.NewGetCollectionsMessage(db.GetCollectionsRequestBody{
+			CheckpointIDs: []string{checkpointID1},
+			Collections:   []string{fmt.Sprintf("%s.%s", collection.ScopeName(), collection.Name())},
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = btc.pushReplication.sendMsg(getCollectionsRequest)
+		if err != nil {
+			return nil, err
+		}
+		resp := getCollectionsRequest.Response()
+		if resp == nil {
+			return nil, fmt.Errorf("error in getCollections response")
+		}
+		btcCollection, err = btc.Collection(scopeAndCollectionKey)
+		if err != nil {
+			return nil, err
+		}
+		_, found := btc.pushReplication.WaitForMessage(getCollectionsRequest.SerialNumber())
+		if !found {
+			return nil, fmt.Errorf("error waiting for getCollections serial number to be stored by the replicator")
+		}
+	} else {
+		btcCollection = btc.DefaultCollection()
+	}
+	return btcCollection, nil
+}
+
+func BlipClientInitialization(t *testing.T, rt *RestTester, collection *db.DatabaseCollection, options *BlipTesterClientOpts) (*BlipTesterClient, error) {
```

Comment:
It looks like all tests require a similar set of steps to get a BLIP test able to work with a single collection, which is going to be the common case.

1. `NewRestTester`
2. `rt.GetSingleTestDatabaseCollection`
3. `BlipClientInitialization(t, rt, collection, nil)`
4. `client.BlipClientCollectionSetup(collection)`

Given that most (all?) tests require these steps and in this order, I'm wondering if it makes sense to at least merge the last two or three like this:

1. `rt := NewRestTester`
2. `client, btcCollection, err := NewBlipClientWithSingleTestCollection(rt)`

So you could leave RestTester creation to the test (given there are different configs required), and then a utility function to give you a blip client and single collection from a RestTester that is passed in, which does the job of `BlipClientInitialization` and `BlipClientCollectionSetup` in one go.

I think maybe this falls apart when you need to use `BlipTesterSpec` but spec could be an optional parameter passed into the constructor.